src/subcommand/server.rs

```diff
@@ -712,6 +717,57 @@ impl Server {
     Ok(Json(hex::encode(metadata)))
   }
 
+  async fn inscription_details(
+    Extension(page_config): Extension<Arc<PageConfig>>,
+    Extension(index): Extension<Arc<Index>>,
+    Path(inscription_id): Path<InscriptionId>,
+  ) -> ServerResult<Response> {
+    let inscription = index
+      .get_inscription_by_id(inscription_id)?
+      .ok_or_not_found(|| format!("inscription {inscription_id}"))?;
+
+    let entry = index
+      .get_inscription_entry(inscription_id)?
+      .ok_or_not_found(|| format!("inscription {inscription_id}"))?;
+
+    let satpoint = index
+      .get_inscription_satpoint_by_id(inscription_id)?
+      .ok_or_not_found(|| format!("inscription {inscription_id}"))?;
+
+    let output = if satpoint.outpoint == unbound_outpoint() {
+      None
+    } else {
+      Some(
+        index
+          .get_transaction(satpoint.outpoint.txid)?
+          .ok_or_not_found(|| format!("inscription {inscription_id} current transaction"))?
+          .output
+          .into_iter()
+          .nth(satpoint.outpoint.vout.try_into().unwrap())
+          .ok_or_not_found(|| format!("inscription {inscription_id} current transaction output"))?,
+      )
+    };
+
+    let address = output
+      .as_ref()
+      .and_then(|o| page_config.chain.address_from_script(&o.script_pubkey).ok())
+      .map(|address| address.to_string());
+
+    Ok(
+      Json(InscriptionDetailsJson {
+        address,
+        inscription_number: entry.inscription_number,
+        content_type: inscription.content_type().map(|s| s.to_string()),
```

Comment:
```suggestion
        type: inscription.content_type().map(|s| s.to_string()),
```

Reply:
@casey @raphjaph "type" is a reserved keyword, so changign this resulted in a bunch of errors. So, I kept `content_type` and `content_length` as I wasn't sure how to change `type`.

Reply:
content_type should be fine